app/src/main/java/com/grofers/hikingTrailsMap/maphiking/MapsActivity.java

```diff
@@ -0,0 +1,477 @@
+                        currentLocation = locationManager.getLastKnownLocation(LocationManager.GPS_PROVIDER);
+                        locationManager.requestLocationUpdates(
+                                LocationManager.GPS_PROVIDER, 5000, 10, locationListener);
+
+                        LatLng latLng = new LatLng(currentLocation.getLatitude(), currentLocation.getLongitude());
+                        trail.add(latLng);
+                        mMap.addMarker(new MarkerOptions().position(latLng).title("Your Location"));
+                        mMap.moveCamera(CameraUpdateFactory.newLatLngZoom(latLng, 16));
+
+
+
+
+                    }catch (SecurityException | NullPointerException e)
+                    {
+                        e.printStackTrace();
+                    }
+
+
+
+                }
+                else
+                {
+                    //Todo: toast can't do without gps??
+                    // permission denied, boo! Disable the
+                    // functionality that depends on this permission.
+
+                }
+
+            }
+
+            // other 'case' lines to check for other
+            // permissions this app might request
+        }
+    }
+
+    @Override
+    public void onLocationChange(Location location) {
+
+        try
+        {
+            //Todo plot on map here
+            Log.i(TAG, "MapsActivity onLocationChange called");
+            System.out.println("changing frequently");
+            //Todo: add location to trail(arraylist<LatLng>)
+            List<LatLng> points  = new ArrayList<>();
+            LatLng currentLatLng = new LatLng(currentLocation.getLatitude(),currentLocation.getLongitude());
+            LatLng newLatLng = new LatLng(location.getLatitude(),location.getLongitude());
+            currentLocation = location;
+            trail.add(newLatLng);
+            points.add(currentLatLng);
+            points.add(newLatLng);
+            PolylineOptions polylineOptions = new PolylineOptions();
+            polylineOptions.addAll(points).color(Color.BLUE).width(5).geodesic(true);
+            mMap.addPolyline(polylineOptions);
+        }
+        catch (NullPointerException e)
+        {
+            e.printStackTrace();
+        }
+
+    }
+
+
+    //Async Tasks
+    private void createTrail()
+    {
+        final ProgressDialog progressDialog;
+        progressDialog = new ProgressDialog(context);
+        progressDialog.setMessage("Creating Your Trail, Please wait...");
+        progressDialog.setIndeterminate(true);
+        progressDialog.show();
+
+        Gson gson = new Gson();
+        List<Double> loc = new ArrayList<>();
+        loc.add(trail.get(0).latitude);
+        loc.add(trail.get(0).longitude);
+        TrailDetail trailDetail = new TrailDetail(trail, androidId, loc);
+        String url = "http://192.168.1.55:5000/trails/create";
+        String jsonString = gson.toJson(trailDetail);
+        JSONObject jsonObject = null;
```

Comment:
can't we convert POJO to JSONObject directly without converting it to String first.